### PR TITLE
[Serverless] skip hostname resolution in Serverless context

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

Skip hostname resolution in serverless context when loading and validation the agent config

### Motivation

Extension start time / cold start improvements

### Describe how to test your changes

Unit testing 

### Checklist

- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/serverless/trace/trace_test.go [hunk 1]
```diff
@@ -64,3 +65,20 @@ func TestStartEnabledTrueValidConfigValidPath(t *testing.T) {
 	assert.NotNil(t, agent.cancel)
 
 }
+
+func TestLoadConfigShouldBeFast(t *testing.T) {
```
Comment (nhinsch) [inline pkg/serverless/trace/trace_test.go:69]: :100:

File: pkg/trace/config/config.go [hunk 2]
```diff
@@ -318,7 +315,20 @@ func Load(path string) (*AgentConfig, error) {
...
+
+// Load returns a new configuration based on the given path. The path must not necessarily exist
+// and a valid configuration can be returned based on defaults and environment variables. If a
+// valid configuration can not be obtained, an error is returned.
+func Load(path string) (*AgentConfig, error) {
+	return load(path, true)
+}
+
+// LoadServerless see Load comment
+// This will skip hostname resolution as hostname need to be empty in a serverless context
```
Comment (nhinsch) [inline pkg/trace/config/config.go:329]: Nit:
```suggestion
// This will skip hostname resolution as hostname is empty in a serverless context
```

**Review by nhinsch — APPROVED** [review]
Looks great! 🥇

File: pkg/trace/config/config.go [hunk 3]
```diff
@@ -197,14 +197,14 @@ func (c *AgentConfig) APIKey() string {
 }
 
 // Validate validates if the current configuration is good for the agent to start with.
-func (c *AgentConfig) validate() error {
+func (c *AgentConfig) validate(hostNameResolution bool) error {
```
Comment (gbbr) [inline pkg/trace/config/config.go:200]: Please don't take this approach. Simply just add a new boolean field to `*AgentConfig` called `DisableGRPCHostname` and return from `acquireHostname` early with an error if it's set:
```go
return errors.new("disabled by config")
```

Comment (gbbr) [reply pkg/trace/config/config.go:200]: Oh, wait. You can't do that.

**Review by gbbr — CHANGES_REQUESTED** [review]
I don't think this is a good approach. Design-wise boolean args are smells and functionality wise I don't think it's a good idea to disable hostname acquisition regardless of situation. The data should be shared between agents.

The right path here is to fix the gRPC client so that it fails early (and doesn't stall by timing out) if the endpoint is not available.

**Comment by maxday** [comment]
Closing this one since this should be a better implementation : https://github.com/DataDog/datadog-agent/pull/8850

**Pull request closed without merging**